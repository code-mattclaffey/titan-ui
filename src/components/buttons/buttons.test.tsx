import React from 'react';
import { render, cleanup } from '@testing-library/react';
import each from 'jest-each';
import { Button, ButtonLink } from '.';

afterEach(() => {
    cleanup();
});

describe('<Button />', () => {
    it('should match snapshot', () => {
        const { container } = render(<Button>Hello World</Button>);

        expect(container).toMatchSnapshot();
    });

    each([
        ['secondary', 'c-button--secondary'],
        ['plain', 'c-button--plain'],
        ['primary', 'c-button--primary'],
        ['outline', 'c-button--outline'],
    ]).test(
        'should render a %s class when %s is set as a prop variant prop',
        (variant, expectedClassName) => {
            const { queryByTestId, container } = render(
                <Button variant={variant}>Hello World</Button>,
            );

            expect(container).toMatchSnapshot();
            expect(
                queryByTestId('c-button').classList.contains(
                    expectedClassName,
                ),
            ).toEqual(true);
        },
    );

    each([
        ['large', 'c-button--large'],
        ['small', 'c-button--small'],
        ['slim', 'c-button--slim'],
    ]).test(
        'should render a %s class when %s is set as a prop size prop',
        (size, expectedClassName) => {
            const { queryByTestId, container } = render(
                <Button size={size}>Hello World</Button>,
            );

            expect(container).toMatchSnapshot();
            expect(
                queryByTestId('c-button').classList.contains(
                    expectedClassName,
                ),
            ).toEqual(true);
        },
    );

    it('should render the wide class when wide prop is added', () => {
        const { queryByTestId, container } = render(
            <Button wide>Hello World</Button>,
        );

        expect(container).toMatchSnapshot();
        expect(
            queryByTestId('c-button').classList.contains(
                'c-button--wide',
            ),
        ).toEqual(true);
    });

    it('should render the wide class when disabled prop is added', () => {
        const { queryByTestId, container } = render(
            <Button disabled>Hello World</Button>,
        );

        expect(container).toMatchSnapshot();
        expect(
            queryByTestId('c-button').classList.contains(
                'c-button--disabled',
            ),
        ).toEqual(true);

        expect(
            queryByTestId('c-button').hasAttribute('disabled'),
        ).toEqual(true);
    });
});

describe('<ButtonLink />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <ButtonLink url="/relative/path">Hello World</ButtonLink>,
        );

        expect(container).toMatchSnapshot();
    });

    each([
        ['secondary', 'c-button--secondary'],
        ['plain', 'c-button--plain'],
        ['primary', 'c-button--primary'],
        ['outline', 'c-button--outline'],
    ]).test(
        'should render a %s class when %s is set as a prop variant prop',
        (variant, expectedClassName) => {
            const { queryByTestId, container } = render(
                <ButtonLink url="/relative/path" variant={variant}>
                    Hello World
                </ButtonLink>,
            );

            expect(container).toMatchSnapshot();
            expect(
                queryByTestId('c-button').classList.contains(
                    expectedClassName,
                ),
            ).toEqual(true);
        },
    );

    each([
        ['large', 'c-button--large'],
        ['small', 'c-button--small'],
        ['slim', 'c-button--slim'],
    ]).test(
        'should render a %s class when %s is set as a prop size prop',
        (size, expectedClassName) => {
            const { queryByTestId, container } = render(
                <ButtonLink url="/relative/path" size={size}>
                    Hello World
                </ButtonLink>,
            );

            expect(container).toMatchSnapshot();
            expect(
                queryByTestId('c-button').classList.contains(
                    expectedClassName,
                ),
            ).toEqual(true);
        },
    );

    it('should render the wide class when wide prop is added', () => {
        const { queryByTestId, container } = render(
            <ButtonLink url="/relative/path" wide>
                Hello World
            </ButtonLink>,
        );

        expect(container).toMatchSnapshot();
        expect(
            queryByTestId('c-button').classList.contains(
                'c-button--wide',
            ),
        ).toEqual(true);
    });

    it('should render the wide class when disabled prop is added', () => {
        const { queryByTestId, container } = render(
            <ButtonLink url="/relative/path" disabled>
                Hello World
            </ButtonLink>,
        );

        expect(container).toMatchSnapshot();
        expect(
            queryByTestId('c-button').classList.contains(
                'c-button--disabled',
            ),
        ).toEqual(true);
    });
});
