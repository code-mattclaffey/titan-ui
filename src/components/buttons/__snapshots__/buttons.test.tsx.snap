// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="c-button c-button--primary"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a large class when c-button--large is set as a prop size prop 1`] = `
<div>
  <button
    class="c-button c-button--primary c-button--large"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a outline class when c-button--outline is set as a prop variant prop 1`] = `
<div>
  <button
    class="c-button c-button--outline"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a plain class when c-button--plain is set as a prop variant prop 1`] = `
<div>
  <button
    class="c-button c-button--plain"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a primary class when c-button--primary is set as a prop variant prop 1`] = `
<div>
  <button
    class="c-button c-button--primary"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a secondary class when c-button--secondary is set as a prop variant prop 1`] = `
<div>
  <button
    class="c-button c-button--secondary"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a slim class when c-button--slim is set as a prop size prop 1`] = `
<div>
  <button
    class="c-button c-button--primary c-button--slim"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render a small class when c-button--small is set as a prop size prop 1`] = `
<div>
  <button
    class="c-button c-button--primary c-button--small"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render the wide class when disabled prop is added 1`] = `
<div>
  <button
    class="c-button c-button--primary c-button--disabled"
    data-testid="c-button"
    disabled=""
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<Button /> should render the wide class when wide prop is added 1`] = `
<div>
  <button
    class="c-button c-button--primary c-button--wide"
    data-testid="c-button"
    type="button"
  >
    Hello World
  </button>
</div>
`;

exports[`<ButtonLink /> should match snapshot 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a large class when c-button--large is set as a prop size prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary c-button--large"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a outline class when c-button--outline is set as a prop variant prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--outline"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a plain class when c-button--plain is set as a prop variant prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--plain"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a primary class when c-button--primary is set as a prop variant prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a secondary class when c-button--secondary is set as a prop variant prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--secondary"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a slim class when c-button--slim is set as a prop size prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary c-button--slim"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render a small class when c-button--small is set as a prop size prop 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary c-button--small"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render the wide class when disabled prop is added 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary c-button--disabled"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;

exports[`<ButtonLink /> should render the wide class when wide prop is added 1`] = `
<div>
  <a
    class="e-link c-button c-button--primary c-button--wide"
    data-testid="c-button"
    href="/relative/path"
  >
    Hello World
  </a>
</div>
`;
